src/p2f/src/CodeOwls.PowerShell/CodeOwls.PowerShell.Provider/Provider.cs

```diff
@@ -1541,14 +1542,15 @@ private IContentWriter DoGetContentWriter(string path)
                 return null;
             }
 
-            return GetContentWriter(path, factories.First(a => a is ISetItemContent));
+            return GetContentWriter(path, factories.FirstOrDefault(a => a is ISetItemContent));
         }
 
         private IContentWriter GetContentWriter(string path, IPathNode pathNode)
         {
             var getContentWriter = pathNode as ISetItemContent;
             if (null == getContentWriter)
             {
+                WriteCmdletNotSupportedAtNodeError(path, ProviderCmdlet.SetContent, SetContentNotSupportedErrorId);
```

Comment:
Please add test cases for happy path, for example,  Set-Content  or New-Item  /$home/foo/t.txt should work  and get-content ~/foo/t.txt, is expected to work.

Reply:
Will do. Thanks!